e2e/e2e.go

```diff
@@ -329,7 +339,9 @@ func main() { //nolint:funlen,gocyclo // Why: there are no reusable parts to ext
 
 	if requireDevconfigAfterDeploy {
 		log.Info().Msg("Running devconfig")
-		runDevconfig(ctx)
+		if err := runDevconfig(ctx); err != nil {
+			log.Fatal().Err(err).Msg("failed to run devconfig")
```

Comment:
I don't like that this function is using a bunch of `Fatal` calls, and also uses several `defer`s. These will be skipped if this is exiting the process directly.

`Fatal` shouldn't be used in any robust function.

In particular, it _really_ seems like this should be a `return` - but that's also true of all the other `Fatal` lines in this function.

Reply:
You're right - I was just following the current conventions of the file. If we want to leave this open for a little bit I can come back to it and refactor the entire file. I won't have time to do it before July 10th realistically though.

Reply:
Can you throw a TODO at the top of this function to replace the `Fatal` logs with something better, noting the `os.Exit` behavior, and call it good?